.gitignore

```diff
@@ -40,16 +40,6 @@ nim_status_client.log
 
 # https://github.com/github/gitignore/blob/master/CMake.gitignore
 CMakeLists.txt.user
-CMakeCache.txt
-CMakeFiles
-CMakeScripts
-Testing
-Makefile
-cmake_install.cmake
-install_manifest.txt
-compile_commands.json
```

Comment:
I know it's a build/configuration product, but some tools (clang-tidy) require it to be in the root of the source dir, I would keep it.